library/x509_crt.c

```diff
@@ -1839,6 +1965,31 @@ int mbedtls_x509_parse_subject_alt_name(const mbedtls_x509_buf *san_buf,
         }
         break;
 
+        /*
+         * RFC822 Name
+         */
+        case (MBEDTLS_ASN1_SEQUENCE | MBEDTLS_X509_SAN_RFC822_NAME):
+        {
+            mbedtls_x509_name rfc822Name;
```

Comment:
I think that it would be best to split this PR into two. This is for parsing an additional subject alt name type, even though the PR name (or description) mentions it. Also, there are no tests for this code.